examples/GettingStarted/BinaryClassification_SentimentAnalysis/SentimentData.cs

```diff
@@ -0,0 +1,13 @@
+﻿using Microsoft.ML.Runtime.Api;
+
+namespace BinaryClassification_SentimentAnalysis
+{
+    public class SentimentData
+    {
+        [Column("0")]
+        public string SentimentText;
+
+        [Column("1", name: "Label")]
+        public float Sentiment;
+    }
+}
```

Comment:
Why does this class needs its own file? You can keep SentimentData, SentimentPrediction and TestSentimentData in Program.cs.

Reply:
The general .NET coding guidelines use one class per file and we want our samples to use idiomatic C#.

Reply:
@OliaG Can you please point me to this guideline? Vast majority of ML.Net code has multiple class definitions in one file.

Reply:
If there isn't a written rule then we should keep things consistent with the current code base. [Here is an example](https://github.com/dotnet/corefx/blob/ddd1e5ce0f81f7091e544c65f99c37bf52a4ded9/src/System.Runtime.CompilerServices.VisualC/ref/System.Runtime.CompilerServices.VisualC.cs) from .Net Corefx repo that shows multiple classes defined in one file.

Reply:
@codemzs 

> If there isn't a written rule then we should keep things consistent with the current code base

I disagree. Samples aren't part of the ML.NET codebase. They are for our customers and the vast majority doesn't need to know or care bout the ML.NET coding conventions. That's why we try to follow the conventions that the vast majority of our customers are using.

That being said, I don't have a problem if we were to merge all these classes into one file as it might even help to keep the sample easier to understand.

Reply:
The guideline is interesting, and one we generally try to obey ourselves, but *not* when separating definitions would lead to less comprehensible code. So we tend to declare types meant to be understood together as a unit, in the same file. (This isn't Java after all. :) ) This clearly falls into that bucket. Especially in this case where the intent is pedagogical, we would "bend" any rules we usually follow about style (even if this was an actual policy, which, as Zeeshan points out, it is not).

In *this* case what I'd rather do is have a (private?) nested type, declared right next to their usages in the method. This will structure the example so that the method and class can be understood. Then you can get rid of these three files, and have only one, more easily understood example.

---
In reply to: [190942775](https://github.com/dotnet/machinelearning/pull/238#discussion_r190942775) [](ancestors = 190942775)